### PR TITLE
Convert some promiscuous code to async/await.

### PR DESCRIPTION
**Pull request by vdbergh**
The async/await framework allows one to write asynchronous code as if it were synchronous. It uses promises under the hood.

### CODE REVIEW

File: server/fishtest/templates/tests_view.mak [hunk 1]
```diff
@@ -559,22 +566,23 @@
...
-        .then((response) => {
-          if (response.ok) return response.text();
-        })
-        .then((text) => {
-          if (!text) return Promise.resolve();
-          return showDiff(text);
-        });
+      });
+      if(response.ok) {
+        var text = await response.text();
```
Comment (peregrineshahin) [inline server/fishtest/templates/tests_view.mak:576]: The deceleration of this variable here will most certainly make it undefined if the response was not ok, resulting in a fatal runtime error, the same goes for the json variable.

better is to declare the variable with a null value outside before the if checking.

Comment (vdbergh) [reply server/fishtest/templates/tests_view.mak:576]: This is just a literal translation of the current code (which has no error handling) to async/await. I would like to keep this as a separate PR. Later we can clean up the code and add error handling, which is easy in the async/await framework. The reason for not doing it now is that it is laborious to test the error handling code paths, and moreover error handling may paper over genuine bugs.

**Comment by ppigazzini** [comment]
LGTM. The code is very clean now, it was a nightmare when I wrote some PRs :)

**Comment by peregrineshahin** [comment]
I notice that the SPSA pre-loader (image/giff) has stopped showing even before this PR, but I'm not sure it is needed from a feature POV.

**Comment by ppigazzini** [comment]
AFTK for a couple of hours, I kept open the PRs for review and test.

**Comment by vdbergh** [comment]
> I notice that the SPSA pre-loader (image/giff) has stopped showing even before this PR not sure when, but I'm not sure it is needed from a feature POV.

I noticed that too. I’ll have a look at (probably not today…, family here).

**Comment by ppigazzini** [comment]
I wonder what is the last warning, it's in all our pages, I don't find that header in the source.

image

**Comment by peregrineshahin** [comment]
It is related to chrome extensions, not Fishtest, I disabled all my chrome extensions and the error disappeared, not sure which one is the culprit in my case though.

**Comment by vdbergh** [comment]
In the latest push there are a few tweaks to make the code slightly more iconic.

**Comment by vondele** [comment]
iconic :-)

**Review by ppigazzini — APPROVED** [review]

**Comment by ppigazzini** [comment]
PROD updated, thank you @vdbergh :)

**Pull request merged**